src/vs/workbench/api/browser/mainThreadTerminalService.ts

```diff
@@ -350,7 +354,7 @@ export class MainThreadTerminalService implements MainThreadTerminalServiceShape
 	}
 
 	private async _onRequestAvailableProfiles(req: IAvailableProfilesRequest): Promise<void> {
-		if (this._isPrimaryExtHost()) {
+		if (this._isPrimaryExtHost() && this._extHostKind !== ExtensionHostKind.LocalWebWorker) {
```

Comment:
do we want to return `this._extHostKind !== ExtensionHostKind.LocalWebWorker` instead of true from `this._isPrimaryExtHost()`?

Reply:
Yes this should go inside `_isPrimaryExtHost` as it probably represents another bug. This is also why we returned `''` in these 2 spots instead of throwing which makes more sense:

https://github.com/microsoft/vscode/blob/3fd2415c9f74eaa82079c2fd1292d7d18341ac3d/src/vs/workbench/api/common/extHostTerminalService.ts#L773-L776

https://github.com/microsoft/vscode/blob/3fd2415c9f74eaa82079c2fd1292d7d18341ac3d/src/vs/workbench/api/common/extHostTerminalService.ts#L790-L792

Reply:
Let's do in debt week https://github.com/microsoft/vscode/issues/119322